agent/config/config_test.go

```diff
@@ -532,6 +540,22 @@ func TestTaskMetadataRPSLimits(t *testing.T) {
 	}
 }
 
+func TestUserDataConfig(t *testing.T) {
```

Comment:
We are missing negative test cases here.

Reply:
added failure cases.